engine/common/rpc/errors.go

```diff
@@ -66,6 +67,27 @@ func ConvertStorageError(err error) error {
 	return status.Errorf(codes.Internal, "failed to find: %v", err)
 }
 
+// convertIndexError converts errors related to index to a gRPC error
```

Comment:
I would recommend to extend godoc of this function to include a mapping of how errors are converted so it's easily accessible.